jmix-multitenancy/multitenancy/src/main/java/io/jmix/multitenancy/core/impl/TenantEntityOperationImpl.java

```diff
@@ -79,9 +84,11 @@ private Field findTenantField(Class<?> entityClass) {
     public void setTenant(Object entity, String tenantId) {
         MetaProperty property = findTenantProperty(entity.getClass());
         if (property != null) {
-            EntityValues.setValue(entity, property.getName(), tenantId);
+            Object reloadedEntity = dataManager.load(Id.of(entity))
```

Comment:
What idea is behind this? Why do we need to reload an instance, modify some field in this **reloaded** instance and then do nothing with this reloaded instance?

Reply:
I chose the wrong way to solve this problem. 
Now, the tenant property will be added to the FetchPlan of the persisted object, if possible.